src/mempool.rs

```diff
@@ -0,0 +1,71 @@
+use std::ops::Index;
+use hyper::{Body, Client, Uri};
+use serde_derive::Deserialize;
+use serde_derive::Serialize;
+
+type ConnectivityResponse = Vec<Nodes>;
+
+#[derive(Default, Debug, Clone, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+struct Nodes {
+    pub public_key: String,
+    pub channels: i64,
+    pub capacity: i64,
+}
+
+#[derive(Default, Debug, Clone, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+struct Node {
+    pub sockets: String,
+}
+
+// TODO return result here
+async fn body_to_string(body: Body) -> String {
+    let body_bytes = hyper::body::to_bytes(body).await.unwrap();
+    String::from_utf8(body_bytes.to_vec()).unwrap()
+}
+
+pub async fn get_recommended_channels() -> Result<Vec<String>, hyper::http::Error> {
+    let https = hyper_tls::HttpsConnector::new();
+    let client = Client::builder()
+        .build::<_, hyper::Body>(https);
+    let uri: Uri = "https://mempool.space/api/v1/lightning/nodes/rankings/connectivity".parse().unwrap();
+    let res = client.get(uri).await.unwrap();
+
+    // Recommend 6 high liquidity channels
+    // Recommend 4 smaller edge nodes
+    let body_str = body_to_string(res.into_body()).await;
+    let high_channels: ConnectivityResponse = serde_json::from_str(&body_str).unwrap();
+    let mut high_cap_nodes = high_channels.clone(); 
+    high_cap_nodes.sort_by(|a, b| a.capacity.partial_cmp(&b.capacity).unwrap());
+    
+    let base_uri = "https://mempool.space/api/v1/lightning/nodes";
+    // Create a vector of lightning addresses
+    // Prefer onion address is accessible
```

Comment:
I'm split on this one, I am thinking we could lean towards IP for the routing node and use onions for the edge nodes

Reply:
Given the tor DDOS attacks I think we should always lead with IP for now